app/renderer/components/tabs/content/closeTabIcon.js

```diff
@@ -50,32 +48,27 @@ class CloseTabIcon extends React.Component {
   mergeProps (state, ownProps) {
     const currentWindow = state.get('currentWindow')
     const frameKey = ownProps.frameKey
-    const frame = frameStateUtil.getFrameByKey(currentWindow, frameKey) || Immutable.Map()
-    const tabId = frame.get('tabId', tabState.TAB_ID_NONE)
-    const isPinnedTab = tabState.isTabPinned(state, tabId)
+    const tabId = frameStateUtil.getTabIdByFrameKey(currentWindow, frameKey)
 
     const props = {}
-    // used in renderer
-    props.showCloseIcon = !isPinnedTab &&
-      (
-        tabContentState.hasRelativeCloseIcon(currentWindow, frameKey) ||
-        tabContentState.hasFixedCloseIcon(currentWindow, frameKey)
-      )
-
-    // used in functions
-    props.frameKey = frameKey
```

Comment:
we need to keep this, because every single ownProps should be passed into new props object